submissions/Harnytskyi/friends-app/js/script.js

```diff
@@ -0,0 +1,151 @@
+const GRID = document.getElementById("allFriends");
+const CHECKBOX_MAN = document.getElementById("gender-male");
+const CHECKBOX_WOMAN = document.getElementById("gender-female");
+const SORT_AZ = document.getElementById("a-z");
+const SORT_ZA = document.getElementById("z-a");
+const SEARCH_INPUT = document.getElementById('search');
+const SORT_09 = document.getElementById("0-9");
+const SORT_90 = document.getElementById("9-0");
+let friends = [];
+
+let selectedFriends = [...friends];
+function filter() {
+    if (gender.male == true && gender.female == true)
+        selectedFriends = friends;
+    else if (gender.male == true && gender.female == false)
+        selectedFriends = friends.filter(item => item.gender == "male")
+    else if (gender.male == false && gender.female == true)
+        selectedFriends = friends.filter(item => item.gender == "female")
+}
+function displayFriends() {
+    GRID.innerHTML = '';
+    selectedFriends.forEach(item => {
+        GRID.innerHTML +=  `<div id="${item.name}" class="card">
+            <img class="photo" src="${item.photo}">
+            <p class="name">${item.name}</p>
+            <p class="age">Age: ${item.age}</p>
+            <a class="phone" href="${item.phone}">${item.phone}</a>
+            <div class="email">
+                <a href="${item.email}">
+                    <button class="button-message">SEND MESSAGE</button>
+                </a>
+            </div>
+        <div>`
+    });
+    search();
+}
+let gender = {
```

Comment:
`let` will not hoist and you have defined it's *after* you have used it.

Bad practice.

Reply:
What does "will not hoist" mean?
Do I just need to move this object higher?

Reply:
https://developer.mozilla.org/en-US/docs/Glossary/Hoisting

`const` and `let` usually will not be visible prior to their declaration. In your case, your function will work fine, because it will get the actual value from the variable during runtime.

So it will work anyway. But this is a good general rule - declaring variables prior they were used :)